Java/core/src/main/java/com/amazon/randomcutforest/state/store/PointStoreMapper.java

```diff
@@ -55,7 +55,12 @@ public PointStore toModel(PointStoreState state, long seed) {
         int[] locationList = new int[indexCapacity];
         int[] tempList = ArrayPacking.unpackInts(state.getLocationList(), state.isCompressed());
         if (!state.getVersion().equals(Version.V3_0)) {
-            System.arraycopy(tempList, 0, locationList, 0, tempList.length);
+            int shingleSize = state.getShingleSize();
+            int baseDimension = dimensions / shingleSize;
+            for (int i = 0; i < tempList.length; i++) {
+                checkArgument(tempList[i] % baseDimension == 0, " incorrect state");
```

Comment:
Can we provide a more descriptive error message?

Reply:
agree, not so easy to get what's wrong with the general error message

Reply:
Done